### PR TITLE
Adding RNNConfig + making RNNForLanguageModeling inherit from PreTrainedModel

### PR DESCRIPTION
**Pull request by Ubadub**
As discussed in our meeting, these changes are to make our custom RNN config class with `save_pretrained` and `from_pretrained`

### CODE REVIEW

File: src/model/rnn.py [hunk 1]
```diff
@@ -266,45 +268,89 @@ def __init__(
         self.output_recurrent_outputs = output_recurrent_outputs
         self.output_last_state = output_last_state
 
+
+# class RNNForLanguageModeling(nn.Module):
```
Comment (shanest) [inline src/model/rnn.py:272]: delete commented line

File: src/model/rnn.py [hunk 2]
```diff
@@ -266,45 +268,89 @@ def __init__(
...
+    _tied_weights_keys = ["lm_head.weight"]
+
+    def __init__(self, config: RNNConfig):
+        super().__init__(config)
+
         self.embedding = nn.Embedding(
             num_embeddings=vocab_size, embedding_dim=embedding_dim, **embedding_kwargs
         )
 
         self.recurrent: nn.RNNBase = getattr(nn, RNNType(rnn_type.upper()))(
```
Comment (shanest) [inline src/model/rnn.py:315]: Where is `rnn_type` defined? Does this need to be `config.rnn_type`? Not obvoius to me how this is working at all otherwise

**Review by shanest — CHANGES_REQUESTED** [review]
looks good!  a couple small things to look at

File: src/model/rnn.py (hunk 2 above)
Comment (Ubadub) [reply src/model/rnn.py]: Good catch, and I thought I had already caught this; turns out I had not pushed that commit, however, which also includes a couple other minor but crucial changes (e.g. the actual config files). Updated now :)

**Review by shanest — APPROVED** [review]
Thanks, looks good now!

**Pull request merged**